src/common/image.c

```diff
@@ -516,16 +555,18 @@ int32_t dt_image_duplicate_with_version(const int32_t imgid, const int32_t newve
       "output_width, output_height, crop, raw_parameters, raw_denoise_threshold, "
       "raw_auto_bright_threshold, raw_black, raw_maximum, "
       "caption, description, license, sha1sum, orientation, histogram, lightmap, "
-      "longitude, latitude, altitude, color_matrix, colorspace, version, max_version, history_end) "
+      "longitude, latitude, altitude, color_matrix, colorspace, version, max_version, history_end, "
+      "position) "
       "SELECT NULL, group_id, film_id, width, height, filename, maker, model, lens, "
       "exposure, aperture, iso, focal_length, focus_distance, datetime_taken, "
       "flags, width, height, crop, raw_parameters, raw_denoise_threshold, "
       "raw_auto_bright_threshold, raw_black, raw_maximum, "
       "caption, description, license, sha1sum, orientation, histogram, lightmap, "
-      "longitude, latitude, altitude, color_matrix, colorspace, NULL, NULL, 0 "
-      "FROM main.images WHERE id = ?1",
+      "longitude, latitude, altitude, color_matrix, colorspace, NULL, NULL, 0, ?1 "
+      "FROM main.images WHERE id = ?2",
       -1, &stmt, NULL);
-  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+  DT_DEBUG_SQLITE3_BIND_INT64(stmt, 1, new_image_position);
```

Comment:
Does that mean that duplicates are always added at the end, instead of next to the original image?

Reply:
Yes, they were. I have fixed that like you have recommended.

Reply:
Done